File: Leetcode/Python/Easy/Longest Common Prefix.py
# https://leetcode.com/problems/longest-common-prefix/

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 1:
            return strs[0]
        else:
            ans = ""
            minimumStr = min(strs, key = len)
            flag = 0
            for j in range(len(minimumStr)):
                for i in range(len(strs)-1):
                    if strs[i][j] == strs[i+1][j]:
                        flag = 1
                    else:
                        flag = 2
                        break
                if flag == 1:
                    ans += strs[i][j]
                elif flag  == 2:
                    break
            return ans

File: Leetcode/Cpp/Easy/Longest Common Prefix.cpp
// https://leetcode.com/problems/longest-common-prefix/

#include<string>
using namespace std;
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() == 1)
        {
            return strs[0];
        }
        else
        {
            string ans = "";
            int minPos, flag = 0;
            for(int i=0; i<=strs.size()-1;i++)
            {
                if(strs[i]<strs[i+1])
                    minPos = i;
            }
            int s = strs[minPos].size()-1;
            for(int i = 0; i<=s; i++)
            {
                for(int j=0; j<=strs.size();j++)
                {
                    if(strs[i][j] == strs[i+1][j])
                    {
                        flag = 1;
                        continue;
                    }
                    else
                    {
                        flag = 0;
                        break;
                    }
                    if(flag == 1)
                        ans += strs[i][j];
                    else if(flag == 0)
                        break;
                }
            }
            return ans;
            
        }
    }
};